### DISCUSSION TITLE
RFC: New (potential) Quirrel.dev Pricing

### DISCUSSION THREAD

**Discussion by Skn0tt** [quirrel-dev/quirrel#70] [Ideas]
Hi there!

I'm thinking about introducing some changes to pricing of https://quirrel.dev (the hosted instance of Quirrel).
Discussion aims to make sure that the new pricing is in-line with what the community would like to see.

> I threw this together really quick, so don't hesitate to give hard feedback! :D

## The current problem:

Currently, pricing is simple: 20$/month for 20.000 jobs.
There's two problems with this:
- All people pay the same, with no regards to the server cost they incur.
- if all you're looking for is a Cron Jobs a day, 20$/month is a lot (and EasyCron is way cheaper).

## My idea for new pricing:

To tackle above-mentioned problems, I'm thinking about adding two new plans to the lineup:

| Calls Per Day | monthly base | 1000 additional calls | |
| - | - | - | - |
| 100 | 10$ | 50c | new, would fit one e.g. one 15-Minute-CronJob |
| 1,000 | 20$ | 25c | same as before, but with 50% more jobs |
| 10,000 | 50$ | 10c | new, for heavy users |

> Having overage billing prevents people with bursty usage patterns from having to switch plans all the time.
> As you may have seen, there's a point where the bigger plan becomes cheaper than paying for overage.
> This aims to incentivise monthly plans, stabilising the reveneue stream.

The Free plan would include 100 API Calls per Day (that's a 30x increase over the current state), but only for non-commercial projects or trying out Quirrel.

> This is new, and it's a measurement to have commercial projects contribute their part to OSS.

The OSS- and Side-Project discounts would still exist.

## What do you think about this?

- What are your thoughts on this?
- What do you like about this pricing?
- What do you dislike?
- Is it too expensive / too cheap / too <fill_in_here> ?

I want these changes to be in-line with community needs. So please, please, please - give feedback below.

**Comment by enricoschaaf** [comment]
For me personally it would be best if there is a free plan with x free calls per month. Everything above the free tier is charged per usage with better pricing for high volume. Custom plans for enterprise users. OSS, side project discounts would be x percentage discounts on the per usage price. The pricing could be significantly higher than competitors due to the way nicer DX. I personally try to avoid using services with fixed monthly pricing and if you are just starting out with a project 10$ are more than you would pay for your managed Postgres instance (Render.com 7$) but if it has to be a fixed price every month maybe a 5$ tier would be an easier sell. ☺️

**Reply by Skn0tt** [reply]
Thanks for your feedback Enrico!

Going fully pay-per-use also sounds interesting. What price points would you like to see there?
(smth like 1k calls cost X$, 10k calls cost Y$, ...)

**Reply by enricoschaaf** [reply]
I think per 1000 calls is probably the most appropriate  for the current demographic that quirrel targets (my guess, you probably have more insides into it), once people use it for multiple projects maybe then you could introduce monthly pricing again, it would be much more compelling for me to pay monthly when I have 1 or more projects running on Quirrel than right at the start.

**Reply by Skn0tt** [reply]
Yeah, that makes sense. You mentioned "better pricing with higher volume" - what kind of progression would you find appropriate? How much should 1k calls be when you buy 1k / 10k / 100k?

**Reply by enricoschaaf** [reply]
I wouldn't decrease the price to much I just checked some other per usage pricing models and I would suggest maybe 25 Cent per 1000 as a base price and above 100k 20 Cent? Just throwing around numbers but when I look at AWS pricing for example the discount is not very massive because that would mean that the base price is too high.

Edit: I also thought about more incremental discounts for 10k and 50k but it does not really make a lot of difference in absolute terms (0.5$-2.5$)

**Comment by ericadamski** [comment]
I would love to be able to give you good, direct feedback. Unfortunately I don't feel like I have a current grasp of my usage. I would love to be able to explore some of my metrics to get a feel for how these limits would affect my current model.

Some feedback without having a baseline, the model you presented looks fine. I would hesitate to use that as substantial input until I am able to see a reference point with my projects. Even if we were to discuss this outside of the quirrel ui.

Hope that was somewhat helpful for now. I would like to provide more info if we can get metrics on my usage currently.

**Comment by HeyGarrison** [comment]
@Skn0tt as you know I'm pretty satisfied with the pricing. However, it seems like other people would prefer usage based pricing. So that might be the best route to go.

My gut tells me that you charging a monthly fee of some kind isn't as important as other businesses, because once someone builds on Quirrel then they're locked in and will most likely take advantage of cron jobs, which in nature provide you with stable monthly revenue. Just my initial thoughts!

**Comment by Skn0tt** [comment]
Thank you all for your helpful replies. 100% usage-based pricing seems like the way to go, so I'll start working on that in the coming weeks :D
